src/Adapter/Order/OrderProductQuantityUpdater.php

```diff
@@ -177,15 +178,52 @@ private function updateOrderDetail(
 
             // Update quantity on the cart and stock
             if ($updateCart) {
-                $updatedProducts = $this->updateProductQuantity($cart, $orderDetail, $oldQuantity, $newQuantity);
-                $this->applyOtherProductUpdates($order, $cart, $orderInvoice, $updatedProducts);
+                $this->setSpecificPriceIfNeeded($cart, $orderDetail);
+                $modifiedProducts = $this->updateProductQuantity($cart, $orderDetail, $oldQuantity, $newQuantity);
+                $this->applyOtherProductUpdates($order, $cart, $orderInvoice, $modifiedProducts);
             }
         }
 
         // Update product stocks
         $this->updateStocks($cart, $orderDetail, $oldQuantity, $newQuantity);
     }
 
+    /**
+     * @param Cart $cart
+     * @param OrderDetail $orderDetail
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function setSpecificPriceIfNeeded(Cart $cart, OrderDetail $orderDetail)
```

Comment:
If you need to use SpecificPrice then it means something is missing along the workflow because OrderDetail should (and must) be enough That's why in https://github.com/PrestaShop/PrestaShop/pull/22299 an new parameter `$keepOrderPrices` was added, the principal behind this parameter is to get the list of product but their prices are fetched from OrderDetail (if the cart is associated to an order of course)

The parameter was added to several functions as there are many possible use cases:
- `getProducts`
- `getOrderTotal`
- `newCalculator`
- `getPackageShippingCost`
- `applyProductCalculations` (protected)

So relying on OrderDetail and this parameter should be enough, however it's more than possible that:
- it has not been used everywhere it should in the Order handlers/services
- it is still missing in some other Cart methods used during computation

Sorry for being picky about this subject but we struggled so long with SpecificPrices that I don't want them to come back in the game 😅  Although it may seem like a good idea at first it results in way too many buggy behaviour (you can see the number of issues associated to the #222999 PR, and it's only the tip of the iceberg), so it's really worth the effort of not using them.